### PR TITLE
FE/IN: Signup / Login Pages

### PR DESCRIPTION
**Pull request by perryraskin**
This PR consists of FE of the signup and login pages, in addition to the integration between the FE and BE. It also contains a server error fix from my previous PR.

Since the header of our website was actually quickly doable by using MaterialUI's `AppBar` component, I added a Header component as well, that acts as our website's menubar as seen in the mockup.

### CODE REVIEW

**Comment by perryraskin** [comment]
Will aim to submit separate PRs for IN in the future 😁✌🏼

**Review by iglesias294 — APPROVED** [review]

**Review by patelparth1993 — APPROVED** [review]

File: client/src/components/Header.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  },
+  button: {
+    backgroundColor: "black",
+    color: "white",
+    fontFamily: 'Poppins',
+    fontWeight: 600,
+    paddingTop: "0.8rem",
+    paddingBottom: "0.8rem",
+    paddingLeft: "2.5rem",
+    paddingRight: "2.5rem",
```
Comment (bonnieli) [inline client/src/components/Header.js]: `padding: 0.8rem 2.5rem` may reduce num lines of code :)

File: client/src/components/Header.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+const useStyles = makeStyles({
+  bar: {
+    backgroundColor: "black",
+    height: "5rem"
+  },
+  title: {
+    marginTop: "60px",
+    marginBottom: "60px",
+    marginLeft: "30px",
+    fontFamily: 'Poppins',
```
Comment (bonnieli) [inline client/src/components/Header.js:21]: could change the entire font Family for the entire app in `theme.js`

File: client/src/pages/Login.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        status = res.status;
+        return res.json();
+      })
+      .then(json => {
+        console.log(json);
+        if (status < 400) setOpenSuccess(true);
+        else setOpenError(true);
+        setResponseMessage(json.message);
+      })
+      .catch(err => {
```
Comment (patelparth1993) [inline client/src/pages/Login.js:97]: Are we going to do client side validation later?

**Pull request merged**